File: agent/query_engine/EngineContext.py
from __future__ import annotations

from dataclasses import dataclass

from llm.chat_message import ChatMessage


@dataclass(frozen=True)
class EngineContext:
    messages: list[ChatMessage]
    user_id: int | None = None
    session_id: int | None = None
    kb_id: int | None = None
    trace_id: str | None = None
    turn_id: str | None = None

File: agent-ts/src/query_engine/EngineContext.ts
import type { ChatMessageDTO } from "../common/model/ChatMessageDTO.js";

export interface EngineContext {
  readonly messages: ChatMessageDTO[];
  readonly userId?: number | null;
  readonly sessionId?: number | null;
  readonly knowledgeBaseId?: number | null;
  readonly traceId?: string | null;
  readonly turnId?: string | null;
}
